src/pages/SettingsPage.js

```diff
@@ -0,0 +1,172 @@
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+import React, { useCallback, useState, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import ValidUrl from 'valid-url';
+import { Text, View, ToastAndroid, TouchableOpacity } from 'react-native';
+import { hashPassword } from 'sussol-utilities';
+import { Button } from 'react-native-ui-components';
+
+import { UIDatabase } from '../database';
+import { SETTINGS_KEYS } from '../settings';
+import { MODAL_KEYS } from '../utilities';
+
+import { gotoRealmExplorer } from '../navigation/actions';
+
+import { ConfirmIcon } from '../widgets/icons';
+import { DataTablePageView, PageInfo } from '../widgets/index';
+import { DataTablePageModal } from '../widgets/modals/index';
+
+import globalStyles from '../globalStyles';
+
+const exportData = async () => {
+  const syncSiteName = UIDatabase.getSetting(SETTINGS_KEYS.SYNC_SITE_NAME);
+  const { success, message } = await UIDatabase.exportData(syncSiteName);
+  const toastMessage = success ? 'Exported data file' : `Couldn't export data: ${message}`;
+  ToastAndroid.show(toastMessage, ToastAndroid.SHORT);
+};
+
+const Settings = ({ toRealmExplorer, currentUser }) => {
+  const [state, setState] = useState({
+    syncURL: UIDatabase.getSetting(SETTINGS_KEYS.SYNC_URL),
+    modalKey: '',
+    syncPassword: '',
+  });
+
+  const { modalKey, syncURL, syncPassword } = state;
+
+  const closeModal = () => setState({ ...state, modalKey: '' });
+  const openModal = newModalKey => setState({ ...state, modalKey: newModalKey });
+  const onSave = () => openModal(MODAL_KEYS.CONFIRM_USER_PASSWORD);
+
+  const editSyncURL = newSyncURL => {
+    if (!ValidUrl.isWebUri(newSyncURL)) ToastAndroid.show('Not a valid URL', ToastAndroid.LONG);
+    else setState({ ...state, modalKey: '', syncURL: newSyncURL });
+  };
+  const editSyncPassword = newSyncPassword =>
+    setState({ ...state, modalKey: '', syncPassword: newSyncPassword });
+
+  const save = enteredPassword => {
+    const currentUserPassowrd = currentUser?.passwordHash ?? '';
```

Comment:
Passowrd??

Reply:
hmm I don't know what the problem is here so I added a comment explaining what a passowrd is f128c10a218f02815b680534ca179d2b3c44bbbf